File: AutomaticTimetableGeneratorSystem-main/timetable-backend/src/main/java/com/timetable/service/TimetableGenerationService.java
package com.timetable.service;

import java.util.List;

import com.timetable.dto.timetablegeneration.GenerateTimetableRequest;
import com.timetable.dto.timetablegeneration.TimetableGenerationResponse;
import com.timetable.dto.timetablegeneration.TimetableSlotDTO;

public interface TimetableGenerationService {

    /**
     * Generate timetable for a specific
     * Academic Year + Course + Semester + Section.
     */
    TimetableGenerationResponse generateTimetable(
            GenerateTimetableRequest request);

    /**
     * Preview generated timetable
     * without saving into database.
     */
    List<TimetableSlotDTO> previewTimetable(
            GenerateTimetableRequest request);

    /**
     * Validate all constraints before generation.
     */
    List<String> validateGeneration(
            GenerateTimetableRequest request);

    /**
     * Delete previously generated timetable.
     */
    void deleteGeneratedTimetable(
            GenerateTimetableRequest request);

    /**
     * Regenerate timetable.
     */
    TimetableGenerationResponse regenerateTimetable(
            GenerateTimetableRequest request);

}

File: AutomaticTimetableGeneratorSystem-main/timetable-frontend/src/services/timetableGenerationService.js
import api from "./api";

const BASE_URL = "/timetable-generation";

const timetableGenerationService = {

    // Generate Timetable
    generateTimetable: async (request) => {

        const response = await api.post(
            `${BASE_URL}/generate`,
            request
        );

        return response.data;

    },

    // Preview Timetable
    previewTimetable: async (request) => {

        const response = await api.post(
            `${BASE_URL}/preview`,
            request
        );

        return response.data;

    },

    // Validate Generation
    validateGeneration: async (request) => {

        const response = await api.post(
            `${BASE_URL}/validate`,
            request
        );

        return response.data;

    },

    // Regenerate Timetable
    regenerateTimetable: async (request) => {

        const response = await api.post(
            `${BASE_URL}/regenerate`,
            request
        );

        return response.data;

    },

    // Delete Generated Timetable
    deleteGeneratedTimetable: async (request) => {

        const response = await api.delete(
            BASE_URL,
            {
                data: request
            }
        );

        return response.data;

    }

};

export default timetableGenerationService;